File: app/src/main/java/com/sakurafish/pockettoushituryou/repository/KindsRepository.java
package com.sakurafish.pockettoushituryou.repository;

import com.sakurafish.pockettoushituryou.model.Kinds;
import com.sakurafish.pockettoushituryou.model.OrmaDatabase;

import java.util.List;

import javax.inject.Inject;

public class KindsRepository {
    private final static String TAG = KindsRepository.class.getSimpleName();

    private final OrmaDatabase orma;

    @Inject
    KindsRepository(OrmaDatabase orma) {
        this.orma = orma;
    }

    public String findName(int kindId) {
        List<Kinds> kinds = orma.relationOfKinds().selector().idEq(kindId).executeAsObservable().toList().blockingGet();
        if (kinds == null || kinds.size() == 0) return "";
        String name = kinds.get(0).name;
        return name != null ? name : "";
    }
}

File: app/src/main/java/com/sakurafish/pockettoushituryou/repository/KindsRepository.kt
package com.sakurafish.pockettoushituryou.repository

import androidx.annotation.WorkerThread
import com.sakurafish.pockettoushituryou.data.db.entity.orma.Kinds
import com.sakurafish.pockettoushituryou.data.db.entity.orma.OrmaDatabase
import javax.inject.Inject

class KindsRepository @Inject internal constructor(private val orma: OrmaDatabase) {

    @WorkerThread
    fun count(): Int {
        return orma.relationOfKinds().count()
    }

    @WorkerThread
    fun findByType(typeId: Int): List<Kinds> {
        return orma.relationOfKinds().selector().typeIdEq(typeId).toList()
    }
}